### DISCUSSION TITLE
Add way to preorder an album before release date

### DISCUSSION THREAD

**Discussion by LouisLouiseKay** [funmusicplace/mirlo#217] [Ideas]
Right now when you set a publish date, if it's not yet released, it just doesn't appear. In comparison this is Bandcamp's system, it's really neat and everyone uses it.

![Screenshot_2023-11-09-21-08-44-564_com android chrome-edit](https://github.com/funmusicplace/mirlo/assets/77789607/e16080c4-49ee-46b1-bb9f-e56d25c98788)

So basically for a pre-order what happens is : the album appears. The previewed songs can be listened to but the rest is greyed out. And if people pay they can download the preview singles.

It'd be amazing to have such a system implemented.

**Comment by simonv3** [comment]
Just want to note that this is already mostly implemented though the logic could take a second look and some edge cases need to be reconsidered.

**Comment by simonv3** [comment]
Closing this because it's implemented (I just pre-ordered an album today) and anything else should be reported as a bug
